plugin/pkg/admission/network/denyserviceexternalips/admission.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package denyserviceexternalips
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	genericadmissioninitializer "k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+const (
+	// PluginName is the name of this admission controller plugin
+	PluginName = "DenyServiceExternalIPs"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		plugin := newPlugin()
+		return plugin, nil
+	})
+}
+
+// externalIPsDenierPlugin holds state for and implements the admission plugin.
+type externalIPsDenierPlugin struct {
+	*admission.Handler
+	lister corev1listers.ServiceLister
+}
+
+var _ admission.Interface = &externalIPsDenierPlugin{}
+var _ admission.MutationInterface = &externalIPsDenierPlugin{}
+var _ = genericadmissioninitializer.WantsExternalKubeInformerFactory(&externalIPsDenierPlugin{})
+
+// newPlugin creates a new admission plugin.
+func newPlugin() *externalIPsDenierPlugin {
+	return &externalIPsDenierPlugin{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+// SetExternalKubeInformerFactory sets a lister and readyFunc for this
+// externalIPsDenierPlugin using the provided SharedInformerFactory.
+func (plug *externalIPsDenierPlugin) SetExternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	informer := f.Core().V1().Services()
+	plug.lister = informer.Lister()
+	plug.SetReadyFunc(informer.Informer().HasSynced)
+}
+
+// ValidateInitialization ensures lister is set.
+func (plug *externalIPsDenierPlugin) ValidateInitialization() error {
+	if plug.lister == nil {
+		return fmt.Errorf("missing lister")
```

Comment:
nit, other plugins seems to provide more information on the error
`return fmt.Errorf("%s requires a service lister", PluginName)`

Reply:
I totally copied this from some other plugin.  I do not know what the best-practice here.